docs/proposals/20200720-single-controller-multitenancy.md

```diff
@@ -0,0 +1,548 @@
+
+* **Management cluster operator**: The cluster operator (ops) is responsible for
+  administration of the Cluster API management cluster. They manage policies, network access,
+  application permissions.
+
+* **Workload cluster operator**: The workload cluster operator (dev) is responsible for
+  management of the cluster relevant to their particular applications .
+
+There are two primary components to the Service APIs security model: RBAC and namespace restrictions.
+
+### RBAC
+RBAC (role-based access control) is the standard used for Kubernetes authorization. This allows 
+users to configure who can perform actions on resources in specific scopes. RBAC can be used to 
+enable each of the roles defined above. In most cases, it will be desirable to have all resources be
+readable by most roles, so instead we'll focus on write access for this model.
+
+##### Write Permissions
+|                              | AzureServicePrincipal, etc | Azure RBAC API | Cluster |
+| ---------------------------- | -------------------- | ----------- | ------- |
+| Infrastructure Provider      | Yes                  | Yes         | Yes     |
+| Management Cluster Operators | Yes                  | Yes         | Yes     |
+| Workload Cluster Operator    | No                   | No          | Yes     |
+
+### Namespace Restrictions
+The extra configuration options are not possible to control with RBAC. Instead,
+they will be controlled with configuration fields on GatewayClasses:
+
+* **allowedNamespaces**: This field is a selector of namespaces that
+  Gateways can use this `AzureServicePrincipal`, `AzureUserAssignedIdentity`, and 
+  `AzureSystemAssignedIdentity` from. This is a standard Kubernetes LabelSelector, a label query 
+  over a set of resources. The result of matchLabels and matchExpressions are ANDed. CAPZ will not 
+  support AzureClusters in namespaces outside this selector. An empty selector (default)
+  indicates that AzureCluster can use this Azure Identity from any namespace. This field is 
+  intentionally not a pointer because the nil behavior (no namespaces) is undesirable here.
+
+
+### CAPZ Controller Requirements
+The CAPZ controller will need to:
+
+* Populate condition fields on AzureClusters and indicate if it is
+  compatible with the Azure Identity. For example, if SystemAssignedIdentity is specified and is not 
+  available, a condition should be set indicating failure.
+* Not implement invalid configuration. For example, if the AzureCluster references an Azure Identity
+  in an invalid namespace for it, it should indicate it through a condition or ignore.
+* Respond to changes in an Azure Identity spec change.
+
+### Risks and Mitigations
+
+#### Caching and handling refresh of credentials
+
+For handling many accounts, the number of calls to the STS service must be minimised. This is 
+currently implemented as a cache on cluster key in Cluster API Provider Azure.
```

Comment:
why cache on cluster instead of credential name/type? would minimize cache duplication and extra calls

Reply:
good idea. I've updated to reflect your feedback.